### ISSUE TITLE
updating attachment causes race condition?

### ISSUE THREAD

**Issue by orenmazor** [thoughtbot/paperclip#2207]
So I'm not super certain of this yet, but its looking like I'm catching a weird race condition.

This may be due to my not using activerecord for my db (using Cequel on cassandra). I'm using s3 storage.

The model:

```
 class Foo
    include Cequel::Record
    include Paperclip::Glue

    has_attached_file :thumb, styles: { original: "300x400>"}, default_url: "/images/:style/missing.png"
    validates_attachment_content_type :thumb, content_type: /\Aimage\/.*\Z/

    column :thumb_file_name, :text
    column :thumb_content_type, :text
    column :thumb_file_size, :text   
    column :width, :int
    column :height, :int
  end
```

The assignment code is equally straightforward, where file is a Tempfile.

```
thumb = file
save
file.unlink
```

I read through the attachment code and it looks like prior to writing the new file, the attributes on the containing model are zeroed out. but it looks like this happens with a callback. which is a problem because so is the saving to s3.

In other words, roughly half the time the callback that zeroes out the attributes happens /after/ we write the new file, deleting the freshly uploaded attachment.

my current workaround is to change the saving code to destroy explicitly, which is kinda crappy. am I missing something?

```
thumb.destroy
save
thumb = file
save
file.unlink
```

I'm on rails 4.2.5, ruby 2.3.1 and paperclip 5beta2

/cc @perobertson

**Comment by caboteria** [comment]
This may describe the same bug as issue #1929

**Comment by sidraval** [comment]
Thank you for reporting this. Unfortunately, we will be deprecating Paperclip and therefore will not have the bandwidth to address this issue.
